.github/workflows/unit_test.yaml

```diff
@@ -1 +1,30 @@
 name: unit test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Setup conda env
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: "latest"
+          activate-environment: test
+          python-version: "3.8"
```

Comment:
yeah I saw that but is it necessary for things like test and linters once the build ci tests for all versions. its going to run x times

Reply:
We need to build from source in the test CI so in this regard "build" and "test" are in the same workflow. 
There's also CI for nightly builds that generates wheels / binaries but the workflow is scheduled, e.g., midnight every day, and typically it includes running unit tests as well.
So I think the CI category is defined not around "build" and "test" but rather "diff time" vs. "nightlies".

For linter it's probably ok to test for just one version -- assuming the linter (and type checker) should remain relatively stable as to their rules.

Reply:
I thought if we have a separate workflow which builds from source for each python, os etc configuration on PRs, its ok to run tests (and linters etc) on a hardcoded configuration? 
or are you suggesting just using this one for testing builds also on PRs (not the recurring build and release workflow)

Reply:
lets do a followup as part of the build CI task

Reply:
RE: separate build workflow --> this I was referring to in my comment as "CI for nightly builds". In addition, there may also be CI for release / stable builds. Those are dedicated build CIs which will test ALL combinations of supported python, os and etc.

For now since we don't do any binary or release builds, "build CI" is simply the build from source step in the test CI before the `pytest` step. So whatever CI matrix we plug in there will be applied to both "build" and "test" steps. 

RE: Avoid hardcoding the configuration: we can [read it as a JSON file](https://stackoverflow.com/questions/61919141/read-json-file-in-github-actions)